ssl-impl/src/main/java/io/crate/protocols/postgres/ssl/SslReqConfiguringHandler.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * This file is part of a module with proprietary Enterprise Features.
+ *
+ * Licensed to Crate.io Inc. ("Crate.io") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ *
+ * Unauthorized copying of this file, via any medium is strictly prohibited.
+ *
+ * To use this file, Crate.io must have given you permission to enable and
+ * use such Enterprise Features and you must have a valid Enterprise or
+ * Subscription Agreement with Crate.io.  If you enable or use the Enterprise
+ * Features, you represent and warrant that you have a valid Enterprise or
+ * Subscription Agreement with Crate.io.  Your use of the Enterprise Features
+ * if governed by the terms and conditions of your Enterprise or Subscription
+ * Agreement with Crate.io.
+ */
+
+package io.crate.protocols.postgres.ssl;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelPipeline;
+import io.netty.handler.ssl.SslContext;
+import io.netty.handler.ssl.SslHandler;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.common.logging.Loggers;
+import org.elasticsearch.common.settings.Settings;
+
+import java.util.function.Supplier;
+
+/**
+ * Handler which configures SSL when it receives an SSLRequest.
+ */
+public class SslReqConfiguringHandler implements SslReqHandler {
+
+    private final Logger LOGGER;
+    private final SslContext sslContext;
+
+    public SslReqConfiguringHandler(Settings settings) {
+        this(settings, new DefaultSslContextSupplier(settings));
+    }
+
+    public SslReqConfiguringHandler(Settings settings, Supplier<SslContext> sslContextSupplier) {
```

Comment:
I think it would now be possible to drop the `Supplier` and require the SslContext directly since it's always created immediately